api/v1beta1/azuremanagedcontrolplane_webhook.go

```diff
@@ -439,6 +441,28 @@ func validateLoadBalancerProfile(loadBalancerProfile *LoadBalancerProfile, fldPa
 	return allErrs
 }
 
+func validateAMCPVirtualNetwork(amcpSpec AzureManagedControlPlaneSpec, fldPath *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+
+	// amcpSpec.VirtualNetwork.ResourceGroup is not set, it means the virtual network is in the same resource group as the control plane.
+	if amcpSpec.ResourceGroupName != amcpSpec.VirtualNetwork.ResourceGroup {
+		amcpVirtualNetwork := amcpSpec.VirtualNetwork
+		// CIDR blocks get defaulted in the defaulting webhook, so we can assume they are always set.
+		_, parentNet, err := net.ParseCIDR(amcpVirtualNetwork.CIDRBlock)
+		if err != nil {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("CIDRBlock"), amcpVirtualNetwork.CIDRBlock, "pre-existing virtual networks CIDR block is wrong"))
```

Comment:
```suggestion
			allErrs = append(allErrs, field.Invalid(fldPath.Child("CIDRBlock"), amcpVirtualNetwork.CIDRBlock, "pre-existing virtual networks CIDR block is invalid"))
```
nit: just a wording choice, feel free to ignore.